test_conformance/api/negative_queue.cpp

```diff
@@ -172,3 +350,259 @@ int test_negative_create_command_queue_with_properties_khr(
+    test_failure_error_ret(
+        err, CL_INVALID_COMMAND_QUEUE,
+        "clGetCommandQueueInfo should return CL_INVALID_COMMAND_QUEUE when: "
+        "\"command_queue is not a valid command-queue\" using a valid object "
+        "that is not a command-queue",
+        TEST_FAIL);
+
+    if (device_supports_on_device_queue(deviceID))
+    {
+        constexpr int NUM_QUEUE_PROPERTIES = 3;
+        const std::array<cl_queue_properties, NUM_QUEUE_PROPERTIES>
+            properties = { CL_QUEUE_PROPERTIES,
+                           CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE, 0 };
+        cl_int err = CL_INVALID_VALUE;
+        clCommandQueueWrapper cmd_queue = clCreateCommandQueueWithProperties(
+            context, deviceID, properties.data(), &err);
+        test_error(err, "clCreateCommandQueueWithProperties");
+        cl_uint queue_size = -1;
+        err = clGetCommandQueueInfo(cmd_queue, CL_QUEUE_SIZE,
+                                    sizeof(queue_size), &queue_size, nullptr);
+        test_failure_error_ret(err, CL_INVALID_COMMAND_QUEUE,
+                               "clGetCommandQueueInfo should return "
+                               "CL_INVALID_COMMAND_QUEUE when: \"command_queue "
+                               "is not a valid command-queue for param_name\"",
+                               TEST_FAIL);
+    }
+
+    constexpr cl_command_queue_info invalid_param = -1;
+    err = clGetCommandQueueInfo(queue, invalid_param, 0, nullptr, nullptr);
+    test_failure_error_ret(
+        err, CL_INVALID_VALUE,
+        "clGetCommandQueueInfo should return CL_INVALID_VALUE when: "
+        "\"param_name is not one of the supported values\"",
+        TEST_FAIL);
+
+
+    cl_uint ref_count = -1;
+    err = clGetCommandQueueInfo(queue, CL_QUEUE_REFERENCE_COUNT, 0, &ref_count,
+                                nullptr);
+    test_failure_error_ret(
+        err, CL_INVALID_VALUE,
+        "clGetCommandQueueInfo should return CL_INVALID_VALUE when: \"size in "
+        "bytes specified by param_value_size is < size of return type and "
+        "param_value is not a NULL value\"",
+        TEST_FAIL);
+
+    return TEST_PASS;
+}
+
+int test_negative_set_command_queue_property(cl_device_id deviceID,
+                                             cl_context context,
+                                             cl_command_queue queue,
+                                             int num_elements)
+{
+    cl_queue_properties property(CL_QUEUE_OUT_OF_ORDER_EXEC_MODE_ENABLE);
+    cl_int err = clSetCommandQueueProperty(nullptr, property, CL_TRUE, nullptr);
+    test_failure_error_ret(
+        err, CL_INVALID_COMMAND_QUEUE,
+        "clSetCommandQueueProperty should return CL_INVALID_COMMAND_QUEUE "
+        "when: \"command_queue is not a valid command-queue\" using a nullptr",
+        TEST_FAIL);
+
+
+    err = clSetCommandQueueProperty(reinterpret_cast<cl_command_queue>(context),
```

Comment:
Think these tests could be over ambitious with the reinterpret cast being an invalid object error on all implementations. There was some discussion [here](https://github.com/KhronosGroup/OpenCL-Docs/issues/482#issuecomment-795685410) about what a valid openCL object is, resulting in merged PR https://github.com/KhronosGroup/OpenCL-Docs/pull/842  that specifies 

> An OpenCL implementation must check for a `NULL` object to determine if an
object is valid.  The behavior for all other invalid objects is implementation-defined.

Reply:
Agreed. Anything other than NULL is implementation defined so we should ideally skip these tests.

Reply:
Done.